pandas/core/generic.py

```diff
@@ -4136,9 +4136,14 @@ def fillna(self, value=None, method=None, axis=None, inplace=False,
                 return result if not inplace else None
 
             elif not is_list_like(value):
-                new_data = self._data.fillna(value=value, limit=limit,
-                                             inplace=inplace,
-                                             downcast=downcast)
```

Comment:
its actually better to add the ``axis=`` parameter to ``.fillna``, a bit more work but removes this logic from here to internals where it belongs. (and the transpose can happen there).